### PR TITLE
Power up unused RAM before entering MCUboot

### PR DESCRIPTION
**Pull request by tomchy**

### CODE REVIEW

File: samples/zigbee/light_switch/src/main.c [hunk 1]
```diff
@@ -555,6 +555,11 @@ static void ota_evt_handler(const struct zigbee_fota_evt *evt)
 
 	case ZIGBEE_FOTA_EVT_FINISHED:
 		LOG_INF("Reboot application.");
+		/* Power on unused sections of RAM to allow MCUboot to use it. */
```
Comment (sebastiandraus) [inline samples/zigbee/light_switch/src/main.c:558]: What do you think about adding dependency to MCUBoot?

File: subsys/zigbee/osif/zb_nrf_platform.c [hunk 2]
```diff
@@ -545,6 +546,11 @@ void zb_reset(zb_uint8_t param)
 	nrf_power_gpregret_set(NRF_POWER, reas);
 #endif /* CONFIG_SOC_NRF5340_CPUAPP */
 
+	/* Power on unused sections of RAM to allow MCUboot to use it. */
```
Comment (sebastiandraus) [inline subsys/zigbee/osif/zb_nrf_platform.c:549]: What do you think about adding dependency to MCUBoot here?

Comment (tomchy) [reply subsys/zigbee/osif/zb_nrf_platform.c:549]: It is not needed. Generally speaking - any bootloader expects to have a full RAM available on cold boot.
If necessary, I may change "MCUboot" to "bootloader", but I do not see the need for that.

**Review by sebastiandraus — APPROVED** [review]

File: subsys/zigbee/osif/zb_nrf_platform.c (hunk 2 above)
Comment (sebastiandraus) [reply subsys/zigbee/osif/zb_nrf_platform.c:549]: Agree, no need to be specific about which bootloader to use.

**Review by MarekPorwisz — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -425,6 +426,10 @@ Bluetooth libraries and services
       * Serialization of the write callback applied to the GATT attribute.
       * Serialization of the :c:func:`bt_gatt_service_unregister` function call.
 
+  * :ref:`lib_ram_pwrdn` - added
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:429]:
```suggestion
  * :ref:`lib_ram_pwrdn`
```

**Review by greg-fer — APPROVED** [review]
Just one comment.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 3 above)
Comment (tomchy) [reply doc/nrf/releases/release-notes-changelog.rst]: Fixed!

**Pull request merged**